File: Micro_Hub_Manager/Backend/gateway/Controllers/taskController.py
from fastapi import APIRouter,Header
from Models.schemas import TaskSchema
import httpx


router = APIRouter(prefix="/taskservice")

SPRING_URL = "http://localhost:8002/"



@router.post("/createtask")
async def signup(T: TaskSchema,token:str = Header(...)):
    async with httpx.AsyncClient() as client:
        response = await client.post(
            SPRING_URL + "task/createtask",
            json=T.model_dump()
        )
    return response.json()

File: Micro_Hub_Manager/Backend/taskservices/controller/taskController.js
import express from "express";
import * as taskService from "../service/taskService.js";
const router = express.Router();

router.post("/createtask", async (req, res) => {
    res.json(await taskService.createTask(req.body, req.headers["token"]));
});

router.get("/getalltasks/:PAGE/:SIZE", async (req, res) => {
    const {PAGE, SIZE} = req.params;
    const response = await taskService.getAllTasks(PAGE, SIZE, req.headers.token);
    res.json(response);
});

export default router;